JWBestPracticeApps/Casting/CastingViewController.swift

```diff
@@ -0,0 +1,181 @@
+//
+//  CastingViewController.swift
+//  Casting
+//
+//  Created by David Almaguer on 9/3/19.
+//  Copyright © 2019 Karim Mourra. All rights reserved.
+//
+
+import UIKit
+import GoogleCast
+
+class CastingViewController: BasicVideoViewController {
+    
+    var avilableDevices: [JWCastingDevice] = []
+    var castController: JWCastController? = nil
+    
+    var castingButton: UIButton? = nil
+    var barButtonItem: UIBarButtonItem? = nil
+    
+    var casting = false {
+        didSet {
+            castingButton?.tintColor = casting ? UIColor.green : UIColor.blue
+        }
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        navigationController?.navigationBar.barTintColor = UIColor.gray
+        
+        // Setup JWCastController object
+        setupCastController()
+    }
+    
+    func setupCastController() {
+        guard let player = self.player else { return }
+        
+        let castController = JWCastController(player: player)
+        castController.chromeCastReceiverAppID = kGCKDefaultMediaReceiverApplicationID
+        castController.delegate = self
+        castController.scanForDevices()
+        self.castController = castController
+    }
+    
+    func setupCastingButton() {
+        let buttonFrame = CGRect(x: 0, y: 0, width: 22, height: 22)
+        let castingButton = UIButton(frame: buttonFrame)
+        castingButton.addTarget(self, action: #selector(castButtonTapped(sender:)), for: .touchUpInside)
+        prepareCastingButtonWithAnimation(castingButton)
+        
+        let barButtonItem = UIBarButtonItem(customView: castingButton)
+        self.navigationItem.rightBarButtonItem = barButtonItem
+        
+        castingButton.heightAnchor.constraint(equalToConstant: 22).isActive = true
+        castingButton.widthAnchor.constraint(equalToConstant: 22).isActive = true
+        
+        self.castingButton = castingButton
+        self.barButtonItem = barButtonItem
+        
+        self.casting = false
+    }
+    
+    func prepareCastingButtonWithAnimation(_ button: UIButton) {
```

Comment:
This method is only called once, how do you feel about moving its implementation into `setupCastingButton` ? I find the name to be confusing because it sounds like an alternative to `setupCastingButton` and not a complement. How about renaming to `addAnimationImages`

Reply:
Yeah I'll move it within the method. As you comment, it's called once and is not used anymore.